src/lib/flow/sol-flow-builder.c

```diff
@@ -581,6 +662,107 @@ get_exported_ports(
     return 0;
 }
 
+static struct sol_flow_node_options *
+builder_type_new_options(const struct sol_flow_node_type *type, const struct sol_flow_node_options *copy_from)
+{
+    struct sol_flow_builder *builder = (struct sol_flow_builder *)type->type_data;
+    struct sol_flow_builder_options *opts;
+    const struct sol_flow_node_options_member_description *member;
+
+    SOL_NULL_CHECK(builder, NULL);
+
+    if (copy_from) {
+        SOL_FLOW_NODE_OPTIONS_API_CHECK(copy_from, SOL_FLOW_NODE_OPTIONS_API_VERSION, NULL);
+        SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(copy_from, SOL_FLOW_BUILDER_API_VERSION, NULL);
+    }
+
+    opts = calloc(1, builder->options_size);
+    SOL_NULL_CHECK(opts, NULL);
+
+    opts->base.api_version = SOL_FLOW_NODE_OPTIONS_API_VERSION;
+    opts->base.sub_api = SOL_FLOW_BUILDER_API_VERSION;
+
+    for (member = type->description->options->members; member->name; member++) {
+        char *dst;
+        const char **src;
+        bool is_string;
+
+        is_string = streq(member->data_type, "string");
+
+        dst = (char *)opts + member->offset;
+        if (copy_from)
+            src = (const char **)((char *)copy_from + member->offset);
+        else
+            src = (const char **)&member->defvalue.ptr;
+
+        if (is_string) {
+            char **s = (char **)dst;
+            free(*s);
+            *s = *src ? strdup(*src) : NULL;
```

Comment:
If `is_string` and `strdup()` fails, will the called know how to handle `*s` being NULL?